packages/fuels-signers/src/provider.rs

```diff
@@ -301,6 +387,51 @@ impl Provider {
         Ok(res)
     }
 
+    /// Returns a vector consisting of `Input::Coin`s and `Input::Message`s for the given
+    /// `ResourceFilter`. The `witness_index` is the position of the witness (signature)
+    /// in the transaction's list of witnesses. In the validation process, the node will
+    /// use the witness at this index to validate the coins returned by this method.
+    pub async fn get_asset_inputs_for_filter(
```

Comment:
```suggestion
    pub async fn get_filtered_inputs(
```

Reply:
I prefer it to be analogous to `get_asset_inputs_for_amount`. The `asset` part is not irrelevant since the returned inputs are always resources. 
If you are still unhappy with it, how about `get_filtered_asset_inputs`

Reply:
I'm fine with `get_filtered_asset_inputs` or even `get_asset_inputs(a_filter_here)` if that name is not taken already.

@hal3e ?

Reply:
I am for `get_asset_inputs(the_filter)` or `get_filtered_asset_inputs`